packages/workflow-core/src/lib/utils/deep-merge-with-options.ts

```diff
@@ -0,0 +1,139 @@
+import { isObject } from '@ballerine/common';
+
+type UnknownRecord = Record<PropertyKey, unknown>;
+
+const mergeObjects = (obj1: UnknownRecord, obj2: UnknownRecord) => {
+  const result = { ...obj1 };
+
+  for (let key in obj2) {
+    if (isObject(obj2[key]) && !Array.isArray(obj2[key]) && key in result) {
+      result[key] = mergeObjects(result[key] as UnknownRecord, obj2[key] as UnknownRecord);
+    } else {
+      result[key] = obj2[key];
+    }
+  }
+
+  return result;
+};
+
+type ArrayOfObjectsWithId = (UnknownRecord & { id: unknown })[];
+
+const mergeArraysById = (arr1: ArrayOfObjectsWithId, arr2: ArrayOfObjectsWithId) => {
+  const combined = [...arr1, ...arr2];
+  const ids = Array.from(new Set(combined.map(item => item.id)));
+
+  return ids.map(id => {
+    const sameIdItems = combined.filter(item => item.id === id);
```

Comment:
you have intersection method on Set object

Reply:
Same